### ISSUE TITLE
Where do we put `NetworkKind`?

### ISSUE THREAD

**Issue by Kixunil** [rust-bitcoin/rust-bitcoin#3238]
I think `NetworkKind` is a different kind (lol) of type than `Network` since it's exhaustive and stable. It's used in multiple places including `PrivateKey`. I'm tempted to put it into `units` but I'm not really sure about it. Any ideas?

**Comment by tcharding** [comment]
If we put it in `units` it smells a bit like "just use `units` because its at the bottom of the stack", perhaps `primitives` is slightly more palatable?

**Comment by Kixunil** [comment]
> "just use `units` because its at the bottom of the stack"

Precisely. But having a completely separate crate for it is stupid. Maybe we should've named `units` differently but no non-stupid name comes to my mind right now.

We can't put it in primitives if we don't want `addresses` to require them, which I certainly don't. If you're only doing frontend verification of the address you shouldn't be forced to pull in `Transaction` and other garbage.

**Comment by apoelstra** [comment]
Yeah let's just throw it in the root of `units`. It's arguably a "boolean unit" if you think of units abstractly enough.
